server/protocols/webadmin/cli/README.md

```diff
@@ -0,0 +1,43 @@
+# Apache James Server/Webadmin command line interface
+
+## Development
+
+'Webadmin command-line interface is an upcoming replacement for the outdated, security-vulnerable JMX command-line interface. It also aims at providing a more modern and intuitive interface.
+It is written in Python using Click - a package that supports creating command-line interfaces. Further information about Click can be found [here](https://click.palletsprojects.com).
+
+
+## Run the script
+1. Enter the following command to get Click installed
+
+      ```$ pip install Click```
+  
+2. Navigate to the location of 'james_cli'.
+
+      Example: ```$ cd Desktop/james-project/james_cli ```
+  
+3. Type ```$ ./james_cli' to show the helping screen ```
+
+
+## Syntax
+
+General syntax to run the script
+
+        ```$ ./james_cli [OPTION] [ENTITY] [ACTION] [ARGUMENT]...```
+where
+
+    [OPTION] optional parameter when running the script,
+  
+    [ENTITY] represents the route that will be implemented,
+  
+    [ACTION] name of the action to perform,
+  
+    [ARGUMENT] argument needed for the action.
+
+Example: ```$ ./james_cli --ip 127.0.0.1 --port 8000 domain list```
```

Comment:
where is the argument? What is optional in fact?

Reply:
My example does not require an argument, it just lists available domains. In case there is an argument, it would be after the sub-command (list, add, remove). In the above example, options --ip and --port using to select the ip address are optional. Without it, the default value will be 127.0.0.1 and 8000 respectively

Reply:
thank you for the explanation. All that you said is very interesting, and it should be in the CLI documentation, not in a github PR comment.